# Chapter 1: Quantities, Units, and Measurement

**Level:** Foundation  \n**Prerequisites:** None  \n**Estimated study time:** 30 minutes

This notebook connects quantities and units to a small production example at Nusantara Rasa Global (NRG), a fictional company. The values are instructional and synthetic.

After completing it, you should be able to represent a measured quantity, convert kilograms to grams, calculate total mass, and check whether the unit of an answer is sensible.

## Situation

A packing line fills 24 cartons. Each carton contains 12 noodle packs, and each pack has a labelled mass of 85 grams. We want the total labelled product mass.

The calculation is:

$$m_{total}=n_{cartons}\times n_{packs/carton}\times m_{pack}$$

The counts have no mass unit. The pack mass is measured in grams, so the result is also in grams.

In [1]:
cartons = 24
packs_per_carton = 12
grams_per_pack = 85

total_grams = cartons * packs_per_carton * grams_per_pack
total_kilograms = total_grams / 1_000

print(f"Total labelled mass: {total_grams:,} g")
print(f"Total labelled mass: {total_kilograms:.2f} kg")

Total labelled mass: 24,480 g
Total labelled mass: 24.48 kg


The line packs 24,480 grams, which is 24.48 kilograms. Dividing by 1,000 changes the unit from grams to kilograms without changing the physical mass.

## Repeated measurements

A label value is not the same as a measured value. The next example uses five synthetic measurements from one pack size.

In [2]:
measurements_g = [84.7, 85.1, 84.9, 85.0, 85.3]
number_measured = len(measurements_g)
measured_total_g = sum(measurements_g)
average_g = measured_total_g / number_measured

print(f"Measurements (g): {measurements_g}")
print(f"Number measured: {number_measured}")
print(f"Total measured mass: {measured_total_g:.1f} g")
print(f"Average measured mass: {average_g:.2f} g")

Measurements (g): [84.7, 85.1, 84.9, 85.0, 85.3]
Number measured: 5
Total measured mass: 425.0 g
Average measured mass: 85.00 g


## Unit check

A useful error check is to write the unit beside every value. Adding 85 grams to 2 cartons is not meaningful because the units describe different kinds of quantities. Multiplying the number of cartons by packs per carton gives packs. Multiplying that result by grams per pack gives grams.

## Practice

Change the example to 30 cartons, 10 packs per carton, and 90 grams per pack. Predict the answer before running the cell. Then report it in grams and kilograms.

In [ ]:
# Write your practice calculation here.
